otherlibs/runtime_events/runtime_events_consumer.c

```diff
@@ -209,9 +213,15 @@ caml_runtime_events_create_cursor(const char_os* runtime_events_path, int pid,
 
   /* This cast is necessary for compatibility with Illumos' non-POSIX
     mmap/munmap */
+#ifndef ATOMIC_LOAD64_RW
   cursor->metadata = (struct runtime_events_metadata_header *)
                       mmap(NULL, cursor->ring_file_size_bytes, PROT_READ,
                           MAP_SHARED, ring_fd, 0);
+#else
+  cursor->metadata = (struct runtime_events_metadata_header *)
+                      mmap(NULL, cursor->ring_file_size_bytes, PROT_READ | PROT_WRITE,
+                          MAP_SHARED, ring_fd, 0);
+#endif
```

Comment:
Could we maybe have the ifdef around a `int prot = ...;` variable, to avoid duplicating the rest of the call? (For example the comment reads a bit weird now as it is farther away from the cast).

Reply:
Actually, I was wondering the same, and also for the `open` call. I've pushed a new implementation.

Reply:
Very nice, thanks!